docs/doc_5_best_practices.md

```diff
@@ -28,6 +28,45 @@ if test != ""
     requestBody.StringPointerValue = test
 ```
 
+## Working with Arrays and Slices
+
+When working with arrays in Golang we need to consider various scenarios such as the absence of a struct property  or property being null.
+Specifically for Atlas API null values aren't allowed which requires developers to specify array objects in specific format.
+Cases for handling arrays for Atlas SDK and API.
+
+1. Update request containing empty array (reset of the field values)
+
+If a struct property is explicitly set to an empty array SDK will send empty array request to the Atlas API.
+
+```golang
+data := Data{
+    dataField: []string{}
+}
+```
+
+2. Update request without updating array field
+
+When performing update operation we recomend the struct property to not present. 
+
+> NOTE: Property definition as nil would trigger error or invalid behaviour
```

Comment:
I am not sure I understand this statement. Does `Property definition as nil` refer to the following example?
```
data := Data{
    dataField: nil
}
```

If this is the case, what would be the risk? Does the marshalling include the property with a `null` value instead of a `[]`?

Reply:
> If this is the case, what would be the risk?

Inconsistency in handling these values in Atlas for updates.

Reply:
> Does the marshalling include the property with a null value instead of a []?

Yes.
Example:
https://go.dev/play/p/mRMMVjdNLwr